internal/controller/components/kueue/kueue_controller.go

```diff
@@ -59,6 +62,17 @@ func (s *componentHandler) NewComponentReconciler(ctx context.Context, mgr ctrl.
 		b = b.WithAction(extraInitialize)
 	}
 
+	labelChangedPredicate := predicate.LabelChangedPredicate{}
+
+	clusterQueueViewerRolePredicate := predicate.Funcs{
```

Comment:
nit: this seems to be a good generic predicate we can eventually move to the `pkg/controller/predicates/resources` package like 

```go
resource.Named(
    types.NamespacedName{Name: "foo"}, 
    types.NamespacedName{Name: "bar"},
)
```